airbyte-ci/connectors/pipelines/pipelines/builds/java_connectors.py

```diff
@@ -18,76 +21,53 @@ class BuildConnectorDistributionTar(GradleTask):
-                "Please review and delete any unnecessary tar files.",
-            )
 
-
-class BuildConnectorImage(BuildConnectorImageBase):
+class BuildConnectorImages(BuildConnectorImagesBase):
     """
-    A step to build a Java connector image using the distTar Gradle task.
+    A step to build Java connector images using the distTar Gradle task.
     """
 
-    async def _run(self, distribution_tar: File) -> StepResult:
+    async def _run(self, dist_dir: Directory) -> StepResult:
+        dist_tar: File
         try:
-            java_connector = await environments.with_airbyte_java_connector(self.context, distribution_tar, self.build_platform)
-            try:
-                await java_connector.with_exec(["spec"])
-            except ExecError:
-                return StepResult(
-                    self, StepStatus.FAILURE, stderr=f"Failed to run spec on the connector built for platform {self.build_platform}."
+            dir_files = await dist_dir.entries()
+            tar_files = [f for f in dir_files if f.endswith(".tar")]
+            num_files = len(tar_files)
+            if num_files != 1:
+                error_message = (
+                    "The distribution tar file for the current java connector was not built."
+                    if num_files == 0
+                    else "More than one distribution tar file was built for the current java connector."
                 )
-            return StepResult(
-                self, StepStatus.SUCCESS, stdout="The connector image was successfully built.", output_artifact=java_connector
-            )
+                return StepResult(self, StepStatus.FAILURE, stderr=error_message)
+            dist_tar = dist_dir.file(tar_files[0])
         except QueryError as e:
             return StepResult(self, StepStatus.FAILURE, stderr=str(e))
+        return await super()._run(dist_tar)
 
-
-class BuildConnectorImageForAllPlatforms(BuildConnectorImageForAllPlatformsBase):
-    """Build a Java connector image for all platforms."""
-
-    async def _run(self, distribution_tar: File) -> StepResult:
-        build_results_per_platform = {}
-        for platform in self.ALL_PLATFORMS:
-            build_connector_step_result = await BuildConnectorImage(self.context, platform).run(distribution_tar)
-            if build_connector_step_result.status is not StepStatus.SUCCESS:
-                return build_connector_step_result
-            build_results_per_platform[platform] = build_connector_step_result.output_artifact
-        return self.get_success_result(build_results_per_platform)
+    async def _build_connector(self, platform: Platform, dist_tar: File) -> Container:
+        return await environments.with_airbyte_java_connector(self.context, dist_tar, platform)
 
 
 async def run_connector_build(context: ConnectorContext) -> StepResult:
     """Create the java connector distribution tar file and build the connector image."""
 
+    if context.use_host_gradle_dist_tar:
```

Comment:
If we move the exception then we can concat the if statement here
```suggestion
    if context.use_host_gradle_dist_tar and context.is_local:
```